frontend/src/components/CreateCredentialModal.tsx

```diff
@@ -4,16 +4,28 @@ import { Fragment } from 'react'
 
 import { useTranslation } from '../lib/acm-i18next'
 export interface ICreateCredentialModalProps {
-  handleModalToggle: () => void
+  control?: any
+  handleModalToggle: (affectedControl: any) => void
 }
 
 export function CreateCredentialModal(props: ICreateCredentialModalProps) {
   const { t } = useTranslation()
+  const { control, handleModalToggle } = props
 
   return (
     <Fragment>
       <ContextSelectorFooter>
-        <Button onClick={props.handleModalToggle} variant="link" isInline>
+        <Button
+          onClick={
+            control
+              ? () => {
+                  handleModalToggle(control)
+                }
+              : handleModalToggle
```

Comment:
You can just pass this function:
```
() => {
  handleModalToggle(control)
}
 ```
 
 If the function doesn't use affectedControl param, passing the argument won't hurt.